ktor-server/ktor-server-servlet-jakarta/jvm/src/io/ktor/server/servlet/jakarta/ServletApplicationEngine.kt

```diff
@@ -20,6 +21,20 @@ import kotlin.coroutines.*
 @MultipartConfig
 public open class ServletApplicationEngine : KtorServlet() {
 
+    override val managedByEngineHeaders: Set<String>
+        get() {
+            servletContext.getAttribute(ApplicationAttributeKey)?.let {
+                return emptySet()
+            }
+
+            val servletContext = servletContext
+            return if ("tomcat" in (servletContext.serverInfo?.toLowerCasePreservingASCIIRules() ?: "")) {
+                setOf(HttpHeaders.TransferEncoding, HttpHeaders.Connection)
+            } else {
+                emptySet()
+            }
+        }
```

Comment:
How about something like...

```kotlin
// Some shared location
fun ServletContext.isTomcat() =
    getAttribute(ApplicationAttributeKey) == null && serverInfo.contains("tomcat", ignoreCase = true)

if (servletContext.isTomcat())
    setOf(HttpHeaders.TransferEncoding, HttpHeaders.Connection)
else emptySet()
```

Reply:
Thank you! It looks much easier :) I'm not sure It can be a shared location because `ktor-server-servlet` and `ktor-server-servlet-jakarta` have different `ServletContext` (from packages `javax.servlet` and `jakarta.servlet`)

Reply:
Ah unfortunate, copy pasta it is 😞